src/sql/base/browser/ui/selectBox/selectBox.ts

```diff
@@ -150,10 +150,10 @@ export class SelectBox extends vsSelectBox implements AdsWidget {
 		this._dialogOptions = options;
 	}
 
-	public selectWithOptionName(optionName?: string, selectFirstByDefault: boolean = true, forceSelectionEvent: boolean = false): void {
+	public selectWithOptionName(optionName?: any, selectFirstByDefault: boolean = true, forceSelectionEvent: boolean = false): void {
```

Comment:
Sorry if I was unclear, but I don't think this is the correct approach either. Select boxes only really make sense with strings so leaving it like this is fine. 

I was suggesting that you fix up `getModelValue` to actually always return a string instead. Right now it says it returns a string, but that's just because we don't have strict type checking on - it's actually returning the original object value (booleans in this case).